Added a regexp check for output returned when using biosdevname,

vamegh opened the pull request:
to see if information returned matches expected interface names from biosdevname.
	modified:   plugins/guests/fedora/cap/configure_networks.rb

tomrogers3 commented:
+1, can we please have this merged. Thanks

cornfeedhobo commented:
Can this please get merged soon. This trips up almost every time I start a vagrant box now.

hnq90 commented:
@mitchellh Please consider to merge this PR.

sethvargo commented:
Hi @vamegh 

Is there a particular issue or bug that this is fixing? Could you please link and explain the problem? Unfortunately this will no longer merge cleanly. Could you please rebase?

vamegh commented:
Hi, 

Sorry for the long time to reply. 

This is / was a bug. Not sure if you guys have fixed it. I detailed it here: 
https://github.com/mitchellh/vagrant/issues/6191

I have updated my branch again.

vamegh commented:
What I have done is this:

```

diff --git a/plugins/guests/fedora/cap/configure_networks.rb b/plugins/guests/fedora/cap/configure_networks.rb
index e4a7e1f..2fd0f96 100644
--- a/plugins/guests/fedora/cap/configure_networks.rb
+++ b/plugins/guests/fedora/cap/configure_networks.rb
@@ -26,6 +26,7 @@ module VagrantPlugins

           if virtual
             machine.communicate.sudo("ls -v /sys/class/net | egrep -v lo\\|docker") do |_, result|
+              next if result !~ /(^en\w\d)|(^sl\w\d)|(^wl\w\d)|(^ww\w\d)|(^em\w\d)|(^p\w\d)|(^eth\d)/
               interface_names = result.split("\n")
             end

@@ -34,12 +35,14 @@ module VagrantPlugins
             end
           else
             machine.communicate.sudo("/usr/sbin/biosdevname -d | grep Kernel | cut -f2 -d: | sed -e 's/ //;'") do |_, result|
+              next if result !~ /(^en\w\d)|(^sl\w\d)|(^wl\w\d)|(^ww\w\d)|(^em\w\d)|(^p\w\d)|(^eth\d)/
               interface_names = result.split("\n")
             end

             interface_name_pairs = Array.new
             interface_names.each do |interface_name|
               machine.communicate.sudo("/usr/sbin/biosdevname --policy=all_ethN -i #{interface_name}") do |_, result|
+                next if result !~ /(^en\w\d)|(^sl\w\d)|(^wl\w\d)|(^ww\w\d)|(^em\w\d)|(^p\w\d)|(^eth\d)/
                 interface_name_pairs.push([interface_name, result.gsub("\n", "")])
               end
             end
```

If you have a better way of fixing it, leave this closed. TBH I havent run into this issue again in a while.

vamegh commented:
Also frankly I should have split off into my own branch rather than directly in my fork master, apologies for that.

attilagyorffy commented:
I am having an issue where a CentOS box (http://puppet-vagrant-boxes.puppetlabs.com/centos-510-i386-virtualbox-puppet.box) fails to enable network interfaces and fails during run:

```
# Bring the interface up
ARPCHECK=no /sbin/ifup ''


Stdout from the command:



Stderr from the command:

usage: ifdown <device name>
Usage: ifup <device name>
```

I wonder if this is a related issue to the bug described here.
